addons/survey/models/survey_survey.py

```diff
@@ -35,14 +35,26 @@ def _default_stage(self):
     page_ids = fields.One2many('survey.page', 'survey_id', string='Pages', copy=True)
     user_input_ids = fields.One2many('survey.user_input', 'survey_id', string='User responses', readonly=True, groups='survey.group_survey_user')
     # security / access
-    auth_required = fields.Boolean('Login required', help="Users with a public link will be requested to login before taking part to the survey")
+    access_mode = fields.Selection([
```

Comment:
Shouldn't we add ir.rules for public and portal, to allow them creating answer on public survey ?

Reply:
Public and portal have no rights. It is simpler to give rights through routes using token; otherwise you have to craft complex ACLs. That way we are sure there is no information leak. Public surveys allow everyone to get a token, and then using that token they can create answers.